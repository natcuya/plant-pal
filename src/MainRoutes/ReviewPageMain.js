import React from 'react'
import Note from '../Note'
import ApiContext from '../ApiContext'
//import './NotePageMain.css'
import config from '../config';

export default class ReviewPageMain extends React.Component {
  state = {
    review: ''
  }
  static defaultProps = {
    match: {
      params: {}
    }
  }
  static contextType = ApiContext

  componentDidMount() {
    const { reviewid } = this.props.match.params
    fetch(`${config.API_ENDPOINT}/reviews/${reviewid}`)
      .then(res => {
        if(!res){
          return res.json().then(e=> Promise.reject(e));
        }
        return res.json();
      })
      .then(review => {
        this.setState({review})
      })
      .catch(error => {
        console.error({error})
      })
  }

  handleDeleteReview = reviewid => {
    console.log('handledeletereview function')
    this.props.history.push(`/`)
  }

  render() {
    let review = this.state.review;
    return (
      <section className='NotePageMain'>
        <Note
          id={review.id}
          rating={review.rating}
          onDeleteReview={this.handleDeleteReview}
        />
        <div className='NotePageMain__content'>
          <p>{review.content}</p>
          {/* {note.content.split(/\n \r|\n/).map((para, i) =>
            <p key={i}>{para}</p>
          )} */}
        </div>
      </section>
    )
  }
}