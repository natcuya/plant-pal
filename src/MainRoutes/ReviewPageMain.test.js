import React from 'react';
import ReactDOM from 'react-dom';
import ReviewPageMain from './ReviewPageMain';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ReviewPageMain />, div);
  ReactDOM.unmountComponentAtNode(div);
});
it('renders the UI as expected', () => {
    const tree = renderer
      .create(<ReviewPageMain name="ReviewPageMain"/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });