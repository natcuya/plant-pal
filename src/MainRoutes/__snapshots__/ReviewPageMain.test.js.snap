// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<section
  className="NotePageMain"
>
  <div
    className="Note"
  >
    <h3
      className="Review_content"
    >
      Review: 
    </h3>
    <h4
      className="Review__ratings"
    >
      Rating: 
    </h4>
  </div>
  <div
    className="NotePageMain__content"
  >
    <p />
  </div>
  <div
    className="ReviewPageNav"
  >
    <button
      className="NavCircleButton ReviewPageNav__back-button"
      onClick={[Function]}
      role="link"
    >
      <br />
      Back
    </button>
  </div>
</section>
`;
